OpenRA.Mods.Common/Activities/RepairBridge.cs

```diff
@@ -16,27 +16,47 @@ namespace OpenRA.Mods.Common.Activities
 {
 	class RepairBridge : Enter
 	{
+		readonly Actor target;
+		readonly LegacyBridgeHut legacyHut;
 		readonly BridgeHut hut;
 		readonly string notification;
 
 		public RepairBridge(Actor self, Actor target, EnterBehaviour enterBehaviour, string notification)
 			: base(self, target, enterBehaviour)
 		{
-			hut = target.Trait<BridgeHut>();
+			this.target = target;
+			legacyHut = target.TraitOrDefault<LegacyBridgeHut>();
+			hut = target.TraitOrDefault<BridgeHut>();
 			this.notification = notification;
 		}
 
 		protected override bool CanReserve(Actor self)
 		{
-			return hut.BridgeDamageState != DamageState.Undamaged && !hut.Repairing && hut.Bridge.GetHut(0) != null && hut.Bridge.GetHut(1) != null;
+			if (legacyHut != null)
+				return legacyHut.BridgeDamageState != DamageState.Undamaged && !legacyHut.Repairing && legacyHut.Bridge.GetHut(0) != null && legacyHut.Bridge.GetHut(1) != null;
+
+			if (hut != null)
+				return hut.BridgeDamageState != DamageState.Undamaged && !hut.Repairing;
+
+			return false;
 		}
 
 		protected override void OnInside(Actor self)
 		{
-			if (hut.BridgeDamageState == DamageState.Undamaged || hut.Repairing || hut.Bridge.GetHut(0) == null || hut.Bridge.GetHut(1) == null)
-				return;
+			if (legacyHut != null)
+			{
+				if (legacyHut.BridgeDamageState == DamageState.Undamaged || legacyHut.Repairing || legacyHut.Bridge.GetHut(0) == null || legacyHut.Bridge.GetHut(1) == null)
+					return;
+
+				legacyHut.Repair(self);
+			}
+			else
```

Comment:
This looks inconsistent with the rest - you're explicitly nullchecking everywhere else instead of an if-else.